kore/src/Kore/Step/Simplification/SubstitutionSimplifier.hs

```diff
@@ -155,7 +167,8 @@ newtype MakeAnd monad =
         { makeAnd
             :: forall variable
             .  InternalVariable variable
-            => TermLike variable
+            => NotSimplifier monad
```

Comment:
Same as for `SubstitutionSimplifier`: if `MakeAnd` needs a `NotSimplifier`, it should capture that dependency when it is created.